pkg/storage/unified/resource/cdk_backend.go

```diff
@@ -146,19 +146,17 @@ func (s *cdkBackend) WriteEvent(ctx context.Context, event WriteEvent) (rv int64
 		})
 	}
 
-	// Async notify all subscribers
+	// notify all subscribers
 	if s.stream != nil {
-		go func() {
-			write := &WrittenEvent{
-				Type:            event.Type,
-				Key:             event.Key,
-				PreviousRV:      event.PreviousRV,
-				Value:           event.Value,
-				Timestamp:       time.Now().UnixMilli(),
-				ResourceVersion: rv,
-			}
-			s.stream <- write
-		}()
+		write := &WrittenEvent{
+			Type:            event.Type,
+			Key:             event.Key,
+			PreviousRV:      event.PreviousRV,
+			Value:           event.Value,
+			Timestamp:       time.Now().UnixMilli(),
+			ResourceVersion: rv,
+		}
+		s.stream <- write
```

Comment:
This will block if there is no other gouroutine reading the channel. Is there a guaranteed reader?

Reply:
Hopefully we get to delete this backend before it's fixed, but for now that sounds like a good idea.

Reply:
@pstibrany https://github.com/grafana/search-and-storage-team/issues/478